### DISCUSSION TITLE
Unwanted movement to some children elements when using translate class.

### DISCUSSION THREAD

**Discussion by Abdulateef-Muhammad** [tailwindlabs/tailwindcss#9910] [Help] [answered]
**What version of Tailwind CSS are you using?**

v3.1.8

**What build tool (or framework if it abstracts the build tool) are you using?**

webpack 5.74.0

**What version of Node.js are you using?**

v16.16.0

**What browser are you using?**

Chrome 107.0.5304.107 (Official Build) (64-bit)

**What operating system are you using?**

Windows 11

**Reproduction URL**
[Link to the repo](https://github.com/Abdulateef-Muhammad/unwated-children-elements-movements-issue)

**Describe your issue**
I'm using -translate-y classes when hover, but on hovering on the my card I got a bit unwanted movement to some children of this card, just like elements shaking a little bit.
I investigated this issue and found the problem is when tailwind generating a transform selector with all its properties values [scale, rotate, translate, ...].
```
transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
```

so I used a custom inline style with one property value of transform and the problem solvoed 
to be like this 
```
transform: translateY(-5px);
```

**Comment by thecrypticace** [comment]
Hey, I'm not seeing this in the project you provided. When using `@tailwind base;` every element has those variables reset so they don't actually cascade / inherit. Are you perhaps not using our base styles / layer?

I creates a slightly simplified version of your example on Tailwind Play (I removed the ratings stars and linked an image from unsplash):
https://play.tailwindcss.com/gJJ9sppIzm

Do you see the problem there?

**Reply by Abdulateef-Muhammad** [reply]
Look at this video from the project I'm working on:

https://user-images.githubusercontent.com/54203350/203710799-38d19da9-2a42-4649-b60a-4986961dc967.mp4


you can see the **"available"** text, and **stars** move a little bit moves horizontally while I used `translate-y-` class on **hover.**
and I found the problem was because tailwind assigns `transform` property all its possible values, and I replaced that with one value `scaleY`, and worked for me.
Any way, my conclusion might not be true, or I might use unproper classes around the card that causes this unwanted movement.

**Comment by general-adhoc** [comment, marked as answer]
It appears this is combination of browser issues and how tailwind applies transforms as the defect appears differently depending on what browser you use.  Chrome on Windows is showing mild horizontal pixel shift on child elements in this example that @thecrypticace posted above:

https://play.tailwindcss.com/gJJ9sppIzm


The solution I found was to apply `will-change-transform` to the div with transforms, which eliminated the unwanted horizontal pixel shifting during transition.  See:
https://play.tailwindcss.com/GPqh1FuEg0

I believe the root cause of this is that the transform is actually applying rotate and skew of 0 in tailwind, and when the browser attempts to apply these to fractional pixel sizes, you get strange behavior.

`transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))`

If you override transform in tailwind to eliminate rotate and skew functions, the undesired shifting will also disappear.  This is likely because the browser has some rounding issues when operating on fractional sizes.

**Reply by Abdulateef-Muhammad** [reply]
> I believe the root cause of this is that the transform is actually applying rotate and skew of 0 in tailwind, and when the browser attempts to apply these to fractional pixel sizes, you get strange behavior.
> 
> `transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y))`

Yeah, I had to use instead of that,  a **custom styled-component CSS**, to get rid of what **transform properties** are applied by _tailwindcss_  like `skew`  and `rotate`.

thank you 👍
